File: valihahotel/reservations/src/main/java/com/valiha/reservation/core/interfaces/factory/CategoryFactory.java
package com.valiha.reservation.core.interfaces.factory;

import com.valiha.reservation.core.entities.models.Category;

public interface CategoryFactory {
  Category create(
    String id,
    String title,
    String type,
    int pax,
    int bigBed,
    int smallBed,
    String image
  );
}

File: website/src/core/entities/factory/CategoryFactory.ts
import { Category } from "@/core/entities/models/Category";
import { CategoryFactory } from "@/core/interfaces/factory/CategoryFactory";

export class CategoryFactoryImpl implements CategoryFactory {
  create(
    id: string,
    title: string,
    type: string,
    pax: number,
    bigBed: number,
    smallBed: number,
    image: string
  ): Category {
    return new Category(id, title, type, pax, bigBed, smallBed, image);
  }
}
